src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs

```diff
@@ -1954,7 +1955,7 @@ private static TimeZoneInfoResult TryGetTimeZoneFromLocalMachine(string id, bool
                 else
                 {
                     value = new TimeZoneInfo(match!._id, match._baseUtcOffset, match._displayName, match._standardDisplayName,
-                                          match._daylightDisplayName, match._adjustmentRules, disableDaylightSavingTime: false);
+                                          match._daylightDisplayName, match._adjustmentRules, disableDaylightSavingTime: false, match.HasIanaId);
```

Comment:
Are both of these changes getting tested with the test below? Is it possible to test both of them?

Reply:
I added a test for the second case too.